pkgs/development/compilers/llvm/common/libclc/find-program.patch

```diff
@@ -0,0 +1,11 @@
+--- a/CMakeLists.txt	2024-08-21 20:41:19.301382341 -0700
++++ b/CMakeLists.txt	2024-08-21 20:41:31.098001312 -0700
+@@ -55,7 +55,7 @@
+   # Import required tools
+   if( NOT EXISTS ${LIBCLC_CUSTOM_LLVM_TOOLS_BINARY_DIR} )
+     foreach( tool IN ITEMS clang llvm-as llvm-link opt )
+-      find_program( LLVM_TOOL_${tool} ${tool} PATHS ${LLVM_TOOLS_BINARY_DIR} NO_DEFAULT_PATH )
++      find_program( LLVM_TOOL_${tool} ${tool} )
```

Comment:
Can you please explain why we need this, ideally in the commit message?

Reply:
If there's a mechanism, but it doesn't work, is that something that we should be fixing upstream?

Reply:
I haven't found a fix which would also not impact the behavior needed for upstream.

Reply:
Ok, I have found a fix that can be upstreamed. Going to begin the process of upstreaming.

Reply:
https://github.com/llvm/llvm-project/pull/105969 PR submitted, will update the drv to use it.

Reply:
Using the PR inside the drv now w/ link to the PR.